File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    let n = Math.max(arr.length, Math.max(...arr.map(v=>v.length)));
    for (let a of arr) while (a.length<n) a.push(0);
    while (arr.length < n) arr.push(Array(n).fill(0));
    return arr;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int rows = arr.length;
        int cols = arr[0].length;
        int max = Math.max(rows, cols);
        int[][] result = new int[max][max];

        for (int i = 0; i < rows; i++) {
            System.arraycopy(arr[i], 0, result[i], 0, cols);
        }
        return result;
    }
}
